### PR TITLE
Simplify + Optimize SharedBlobCacheService a little more

### PR DESCRIPTION
**Pull request by original-brownbear**
We can get some additional savings in code size and complexity on the hot path by adjusting method parameters that are effectively int to int. Also, it makes more sense to use relative positioning in the IO.read/write methods than to use absolute positioning in the shared file. This saves some instructions and makes the code easier to follow by not spreading the logic around the absolute cache file position as much.
Also, this makes the fact that region sizes are effectively limited to `Integer.MAX_VALUE` explicit in code.

### CODE REVIEW

File: x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/shared/SharedBytes.java [hunk 1]
```diff
@@ -256,50 +245,49 @@ public final class IO {
...
         }
 
-        private void checkOffsets(long position, long length) {
-            long pageEnd = pageStart + regionSize;
-            if (position < pageStart || position > pageEnd || position + length > pageEnd) {
-                assert false;
-                throw new IllegalArgumentException("bad access");
+        private void checkOffsets(int position, int length) {
+            if (position < 0 || position + length > regionSize) {
+                offsetCheckFailed();
```
Comment (original-brownbear) [inline x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/shared/SharedBytes.java:283]: just a little voodoo, maybe helps inlining once in a blue moon :)

**Review by romseygeek — APPROVED** [review]
LGTM

**Comment by original-brownbear** [comment]
@elasticmachine update branch

**Comment by original-brownbear** [comment]
@elasticmachine update branch

**Pull request merged**